packages/design-system-react-native/src/components/Checkbox/Checkbox.tsx

```diff
@@ -34,24 +32,19 @@ export const Checkbox = forwardRef<{ toggle: () => void }, CheckboxProps>(
     },
     ref,
   ) => {
-    // Internal state for uncontrolled
-    const [internalSelected, setInternalSelected] = useState(isSelected);
```

Comment:
We’re removing the internal state from the Checkbox component so that it becomes a fully controlled component. This means that the engineer using Checkbox is now responsible for managing its selected state via the isSelected prop. This approach provides clearer state ownership, avoids potential sync issues between internal and external state, and gives consumers more flexibility to handle complex UI scenarios (like form libraries, custom logic, etc.)